apps/backend/src/api/embedded-wallets/use-cases/transfer/index.ts

```diff
@@ -57,17 +56,35 @@ export class Transfer extends UseCaseBase implements IUseCaseHttp<ResponseSchema
 
   async handle(payload: RequestSchemaT): Promise<ResponseSchemaT> {
     const validatedData = this.validate(payload, RequestSchema)
+
+    // Get user data
     const { email } = validatedData
 
     const user = await this.userRepository.getUserByEmail(email, { relations: ['passkeys'] })
+
     if (!user) {
       throw new ResourceNotFoundException(messages.USER_NOT_FOUND_BY_EMAIL)
     }
 
+    if (!user.contractAddress) {
+      throw new ResourceNotFoundException(messages.USER_DOES_NOT_HAVE_WALLET)
+    }
+
     if (!user.passkeys.length) {
       throw new ResourceNotFoundException(messages.USER_DOES_NOT_HAVE_PASSKEYS)
     }
 
+    // Get asset contract address from db
+    const asset = await this.assetRepository.getAssetByCode(validatedData.asset)
```

Comment:
```suggestion
    const asset = await this.assetRepository.getAssetByCode(validatedData.asset)
```

Same here, this check will not be required for NFTs